addons/fleet/wizard/fleet_vehicle_send_mail.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models, _
+
+
+class FleetVehicleSendMail(models.TransientModel):
+    _name = 'fleet.vehicle.send.mail'
+    _inherit = 'mail.composer.mixin'
+    _description = 'Send mails to Drivers'
+
+    vehicle_ids = fields.Many2many('fleet.vehicle', string='Vehicles', required=True)
+    author_id = fields.Many2one('res.partner', 'Author', required=True, default=lambda self: self.env.user.partner_id.id)
+    template_id = fields.Many2one(domain=lambda self: [('model_id', '=', self.env['ir.model']._get('fleet.vehicle').id)])
+    attachment_ids = fields.Many2many(
+        'ir.attachment', 'fleet_vehicle_mail_compose_message_ir_attachments_rel',
+        'wizard_id', 'attachment_id', string='Attachments')
+
+    @api.onchange('template_id')
+    def _onchange_template_id(self):
+        self.attachment_ids = self.template_id.attachment_ids
+
+    def action_send(self):
+        self.ensure_one()
+        without_emails = self.vehicle_ids.driver_id.filtered(lambda a: not a.email)
+        if without_emails:
+            return {
+                'type': 'ir.actions.client',
+                'tag': 'display_notification',
+                'params': {
+                    'type': 'danger',
+                    'message': _("The following vehicle drivers are missing an email address: %s.", ', '.join(without_emails.mapped(lambda a: a.name))),
+                }
+            }
+
+        for vehicle in self.vehicle_ids:
+            vehicle.message_post(
+                author_id=self.author_id.id,
+                body=self.body,
+                email_layout_xmlid='mail.mail_notification_light',
+                message_type='comment',
+                partner_ids=vehicle.driver_id.ids,
+                subject=self.subject,
+            )
+
+    def action_save_as_template(self):
+        model = self.env['ir.model']._get('fleet.vehicle')
+        template_name = _("Vehicle: Mass mail drivers")
+        for record in self:
+            template = self.env['mail.template'].create({
+                'name': template_name,
+                'subject': record.subject or False,
+                'body_html': record.body or False,
+                'model_id': model.id,
+                'use_default_to': True,
+            })
+
+            if record.attachment_ids:
+                attachments = record.env['ir.attachment'].sudo().browse(record.attachment_ids.ids).filtered(lambda a: a.create_uid.id == record._uid)
+                if attachments:
+                    attachments.write({'res_model': template._name, 'res_id': template.id})
+                template.attachment_ids |= record.attachment_ids
+
+            record.write({'template_id': template.id})
+
+            return {
+                'type': 'ir.actions.act_window',
+                'view_mode': 'form',
+                'res_id': template.id,
+                'res_model': 'mail.template',
+                'target': 'new',
+            }
```

Comment:
This seems inconsistent. From your `for record in self`, I assume this method can manipulate multiple wizards (which I don't think is the case). But then, you return on the first iteration, so processing a single record. I guess you can just drop the for loop and it should be good